### PR TITLE
Add howard package

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Now that Marion's core features have been developed, we need it to generate certificates that have a real-life relevance.

## Proposal

We decided to develop FUN-specific certificate in a separated python package, but developed in the same repository. We called this package howard.

- [x] add howard certificates package
- [x] add template debugging view
- [x] add tests for the debugging view
- [x] add tests for the custom fetcher

### CODE REVIEW

File: src/marion/marion/certificates/issuers/base.py [hunk 1]
```diff
@@ -319,13 +324,15 @@ def create(self):
...
-        css = CSS(string=self.get_css().render(self.context), font_config=font_config)
+        html = HTML(string=html_str)
+        css = CSS(string=css_str, font_config=font_config)
 
         certificate = html.render(stylesheets=[css], font_config=font_config)
         certificate.metadata = self.get_metadata()
         certificate.write_pdf(target=certificate_path, zoom=1)
 
-        return certificate_path
+        return certificate
```
Comment (Sifflex) [inline src/marion/marion/certificates/issuers/base.py:338]:
```suggestion
        return certificate_path
```

Comment (quitterie-lcs) [reply src/marion/marion/certificates/issuers/base.py]: Change has been implemented

**Comment by quitterie-lcs** [comment]
Context of `RealisationCertificate` was incomplete and inaccessible (because of some typos errors: mismatch of `RealisationCertificate` called as `RealizationCertificate`).
`Realisation` naming is now the conventional naming (in code and docstrings).

File: src/howard/howard/tests/issuers/test_realisation.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                "position": "CEO",
+                "last_name": "Agarwal",
+                "first_name": "Anant",
+            },
+            "location": "Boston, MA",
+        },
+        scope = "action de formation par apprentissage"
+    )
+
+    assert context == test_context
```
Comment (jmaupetit) [inline src/howard/howard/tests/issuers/test_realisation.py:80]: A carriage return is missing.

File: src/marion/setup.cfg [hunk 3]
```diff
@@ -87,4 +87,4 @@ python_files =
     test_*.py
     tests.py
 testpaths =
-    marion
+    marion
```
Comment (jmaupetit) [inline src/marion/setup.cfg:90]: Ditto.

File: src/howard/howard/templates/marion/certificates/realisation.html [hunk 4]
```diff
@@ -88,7 +88,7 @@ <h1>CERTIFICAT DE REALISATION</h1>
               Fait à : <span>{{ course.organization.location }}</span>
             </div>
             <div class="date">
-              Le : <span>{{ creation_date | date:"SHORT_DATE_FORMAT" }}</span>
+              Le : <span>{{ creation_date }}</span>
```
Comment (jmaupetit) [inline src/howard/howard/templates/marion/certificates/realisation.html:91]: An extra trailing space has been inserted.

File: src/howard/howard/templates/marion/certificates/realisation.html [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      {{ css }}
+    </style>
+  </head>
+  {% endif %}
+  <body>
+    <div class="certificate realisation">
+      <header>
+        <div class="logos">
+          <div class="ministry">
+            <!-- FIXME -->
```
Comment (jmaupetit) [inline src/howard/howard/templates/marion/certificates/realisation.html:17]: :thinking:

File: src/howard/howard/templates/marion/certificates/realisation.html [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            <img src="{% static "howard/logo-ministere-travail.svg" %}" />
+          </div>
+          <div class="organization">
+            <img src="https://upload.wikimedia.org/wikipedia/commons/8/8f/EdX.svg" />
```
Comment (jmaupetit) [inline src/howard/howard/templates/marion/certificates/realisation.html:21]: Seems fragile. Should add a static file.

File: src/howard/howard/templates/marion/certificates/realisation.html [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+              Cachet et signature du responsable du dispensateur de formation
+            </p>
+            <div class="manager">
+              <span class="gender">{{ course.session.manager.gender }}</span>
+              <span class="last-name">{{ course.session.manager.last_name }}</span>
+              <span class="first-name">{{ course.session.manager.first_name }}</span>,
+              <span class="first-name">{{ course.session.manager.position }}</span>
+              <br/>
+
+              <img src="{% static "howard/test-signature.png" %}" />
```
Comment (jmaupetit) [inline src/howard/howard/templates/marion/certificates/realisation.html:105]: This file is missing from static files.

File: src/howard/howard/tests/issuers/test_realisation.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        },
+        student={
+            "gender": "Mme",
+            "first_name": "Jane",
+            "last_name": "Doe",
+            "organization": {"name": "ACME inc."},
+        },
+    )
+
+    assert context == expected
```
Comment (jmaupetit) [inline src/howard/howard/tests/issuers/test_realisation.py:96]: Tests for JSON schema are missing.

File: src/howard/setup.cfg [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+keywords = Django, LMS, learning, PDF
+classifiers =
+    Development Status :: 4 - Beta
+    Framework :: Django
+    Framework :: Django :: 3.1
+    Intended Audience :: Developers
+    License :: OSI Approved :: MIT License
+    Natural Language :: English
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.8
```
Comment (jmaupetit) [inline src/howard/setup.cfg:20]: Consider upgrading to python 3.9

File: src/howard/setup.py [hunk 10]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env python
+"""Setup file for the marion module. All configuration stands in the setup.cfg file."""
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/howard/setup.py:2]: Well spotted! :1st_place_medal:

File: src/marion/marion/certificates/views.py [hunk 11]
```diff
@@ -29,3 +34,63 @@ def create(self, request, *args, **kwargs):
...
+    """
+    issuer_path = request.GET.get("issuer", None)
+    context = request.GET.get(
+        "context",
+        {
+            "course": {
+                "name": "edX Demonstration Course",
+                "session": {
+                    "date": {
+                        "to": datetime.datetime(2020, 3, 1, 10, 0),
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/marion/marion/certificates/views.py]: I've removed hard-coded context used to develop howard's templates.

File: src/marion/marion/certificates/views.py [hunk 12]
```diff
@@ -29,3 +34,63 @@ def create(self, request, *args, **kwargs):
...
+                        "from": datetime.datetime(2020, 2, 1, 23, 0),
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/marion/marion/certificates/views.py]: ditto

File: src/howard/setup.cfg (hunk 9 above)
Comment (jmaupetit) [reply src/howard/setup.cfg:20]: Not for now as not all openfun projects that will integrate Marion are running this version.

File: src/marion/marion/certificates/static/marion/noun_Check_3612574.svg [hunk 13]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 1024 1280" x="0px" y="0px"><title>checked</title><path d="M685.1 381.9c-7.8-7.8-18.3-12.2-29.3-12.2s-21.5 4.3-29.3 12.2l-173.3 172.4-55.5-55.9c-16-16-42.3-16.2-58.5-0.3-16.3 15.9-16.5 42.2-0.5 58.6l84.8 85.4c7.8 7.8 18.3 12.2 29.4 12.2 11 0 21.4-4.3 29.2-12.1l202.8-201.6c16-16 16.2-42.3 0.3-58.5l-0.1-0.2z"/><path d="M973.9 317c-25.3-59.7-61.5-113.3-107.5-159.4-46.1-46-99.7-82.2-159.4-107.5-61.8-26.2-127.4-39.5-195-39.5-276.4 0.1-501.3 225-501.3 501.4s224.9 501.3 501.3 501.3c276.4 0 501.3-224.9 501.3-501.3 0-67.6-13.2-133.2-39.4-195zM512 930.4c-230.7 0-418.4-187.7-418.4-418.4s187.7-418.4 418.4-418.4 418.4 187.7 418.4 418.4-187.7 418.4-418.4 418.4z"/><text x="0" y="1039" fill="#000000" font-size="5px" font-weight="bold" font-family="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif">Created by shuai tawf</text><text x="0" y="1044" fill="#000000" font-size="5px" font-weight="bold" font-family="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif">from the Noun Project</text></svg>
```
Comment (quitterie-lcs) [inline src/marion/marion/certificates/static/marion/noun_Check_3612574.svg:1]: Is a newline necessary for an `svg` file?

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**